### PR TITLE
Break up the main run function

### PR DESCRIPTION
**Pull request by vladsavelyev**
* Break up the long `run` functions into subfunctions to make the code more organized.
* Return a `RunResult` object instead of a dictionary.

Part of https://github.com/MultiQC/MultiQC/issues/1051
Depends on https://github.com/MultiQC/MultiQC/pull/2445

### CODE REVIEW

File: multiqc/multiqc.py [hunk 1]
```diff
@@ -362,33 +395,195 @@ def run(
...
+
+    logger.info("MultiQC complete")
+
+    # Move the log file into the data directory
+    log.move_tmp_log(logger)
+
+    return RunResult(sys_exit_code=sys_exit_code)
+
+
+def _set_up_logging(verbose, quiet, no_ansi):
```
Comment (ewels) [inline multiqc/multiqc.py:528]: Might make sense to move this to [`multiqc/utils/log.py`](https://github.com/MultiQC/MultiQC/blob/main/multiqc/utils/log.py) - there's already the `init_log` function there.

File: multiqc/multiqc.py [hunk 2]
```diff
@@ -362,33 +395,195 @@ def run(
...
     )
-    console.print(
-        f"\n  [dark_orange]///[/] [bold][link=https://multiqc.info]MultiQC[/link][/] :mag: [dim]| v{config.version}\n"
-    )
-    logger.debug(f"This is MultiQC v{config.version}")
 
+    return console
+
+
+def _init(
```
Comment (ewels) [inline multiqc/multiqc.py:546]: I find this a bit too close to the special `__init__`. Can we name it something different?

Comment (ewels) [reply multiqc/multiqc.py:546]: This could probably also be moved, into the `config.py` file and renamed `init_config` or something, for example.

**Review by ewels — APPROVED** [review]
Looks good! Couple of minor comments but feel free to merge.

I feel like quite a lot of these functions can now move into separate files to break up this scripts so that's not so long and a bit easier to navigate. But happy to leave that to a future PR if you like.

File: multiqc/multiqc.py [hunk 3]
```diff
@@ -362,33 +395,195 @@ def run(
...
+
+        _export_sources()
+
+        _write_json_dump()
+
+        _write_html_and_data(
+            tmp_dir=tmp_dir,
+            template_mod=template_mod,
+            filename=filename,
+        )
```
Comment (ewels) [inline multiqc/multiqc.py:489]: Love how clear this is! ideally `multiqc.py` can end up being split up into functions in all the relevant sub-files, and this block can go into `__main__.py` or somewhere, and we can finally get rid of `multiqc.multiqc` megascript :)

File: multiqc/multiqc.py (hunk 1 above)
Comment (vladsavelyev) [reply multiqc/multiqc.py:528]: Yep, I've actually done that later in https://github.com/MultiQC/MultiQC/pull/2442

Comment (ewels) [reply multiqc/multiqc.py:528]: Ok cool - good to merge then 👍🏻

**Comment by ewels** [comment]
I'll leave it to you to merge as it's complex branching.

**Pull request merged**